src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -572,17 +709,110 @@ private static void SendObject(Stream receivingStream, object jsonObject)
+        public virtual void NotifyVirtualMethodOverride()
+        {
+            this.notificationTcs.SetResult($"base {nameof(NotifyVirtualMethodOverride)}");
+        }
+
+        [JsonRpcMethod("base/NotifyVirtualMethodNoOverride")]
+        public virtual void NotifyVirtualMethodNoOverride()
+        {
+            this.notificationTcs.SetResult($"base {nameof(NotifyVirtualMethodNoOverride)}");
+        }
     }
 
-    public class Server : BaseClass
+    /// <summary>
+    /// This class is invalid because a derived method has a different <see cref="JsonRpcMethodAttribute" /> value.
+    /// </summary>
+    public class InvalidOverrideServer : BaseClass
+    {
+        [JsonRpcMethod("child/InvokeVirtualMethodOverride")]
+        public override string InvokeVirtualMethodOverride() => $"child {nameof(InvokeVirtualMethodOverride)}";
+    }
+
+    /// <summary>
+    /// This class is invalid because overloaded methods have different <see cref="JsonRpcMethodAttribute" /> values.
+    /// </summary>
+    public class ConflictingOverloadServer : BaseClass
+    {
+        [JsonRpcMethod("test/string")]
+        public string InvokeOverloadConflictingMethodAttribute(string test) => $"conflicting string: {test}";
+
+        [JsonRpcMethod("test/int")]
+        public string InvokeOverloadConflictingMethodAttribute(string arg1, int arg2) => $"conflicting string: {arg1} int: {arg2}";
+    }
+
+    /// <summary>
+    /// This class is invalid because an overloaded method is missing <see cref="JsonRpcMethodAttribute" /> value.
+    /// The method missing the attribute comes before the method with the attribute.
+    /// </summary>
+    public class MissingMethodAttributeOverloadBeforeServer : BaseClass
+    {
+        public string InvokeOverloadConflictingMethodAttribute(string test) => $"conflicting string: {test}";
+
+        [JsonRpcMethod("test/string")]
+        public string InvokeOverloadConflictingMethodAttribute(string arg1, int arg2) => $"conflicting string: {arg1} int: {arg2}";
+    }
+
+    /// <summary>
+    /// This class is invalid because an overloaded method is missing <see cref="JsonRpcMethodAttribute" /> value.
+    /// The method missing the attribute comes after the method with the attribute.
+    /// </summary>
+    public class MissingMethodAttributeOverloadAfterServer : BaseClass
     {
-        private readonly TaskCompletionSource<string> notificationTcs = new TaskCompletionSource<string>();
+        [JsonRpcMethod("test/string")]
+        public string InvokeOverloadConflictingMethodAttribute(string test) => $"conflicting string: {test}";
 
+        public string InvokeOverloadConflictingMethodAttribute(string arg1, int arg2) => $"conflicting string: {arg1} int: {arg2}";
+    }
+
+    /// <summary>
+    /// This class is invalid because two different methods in the same class have the same <see cref="JsonRpcMethodAttribute" /> value.
+    /// </summary>
+    public class SameAttributeUsedOnDifferentMethodsServer : BaseClass
```

Comment:
Thanks for raising that. That is cause for concern, IMO. I wouldn't have anticipated that from the attribute name or the documentation you had on it before. So at minimum we'd want the docs to be clear on this point. But I think unless someone has studied these updated docs, the expected behavior would be that the original method name no longer works. And fixing that later would be a protocol breaking change. So I think we should fix that now.